spec/services/service_offering_spec.rb

```diff
@@ -0,0 +1,44 @@
+describe ServiceOffering do
+  let(:api_instance) { double(:api_instance, :show_service_offering => service_offering) }
+  let(:service_offering) { ServiceOffering.new(params) }
+  let(:params) { {'service_offering_ref' => '10'} }
+
+  it "#{described_class}#find" do
+    with_modified_env TOPOLOGY_SERVICE_URL: 'http://www.example.com' do
+      allow(service_offering).to receive(:api_instance).and_return(api_instance)
+      expect(described_class).to receive(:new).with({}).and_return(service_offering)
+      ServiceOffering.find(1)
+    end
+  end
+
+  it "#show" do
+    with_modified_env TOPOLOGY_SERVICE_URL: 'http://www.example.com' do
+      allow(service_offering).to receive(:api_instance).and_return(api_instance)
```

Comment:
I think this `with_modified_env` and `allow` block can be moved similarly to my above comment.